lib/beaker/dsl/helpers/host_helpers.rb

```diff
@@ -550,15 +550,14 @@ def file_contents_on(host, file_path)
           file_contents = nil
 
           split_path = win_ads_path(file_path)
-
           if file_exists_on(host, split_path[:path])
             if host['platform'] =~ /windows/
               file_path.gsub!('/', '\\')
 
               command = %{Get-Content -Raw -Path #{file_path}}
               command += %{ -Stream #{split_path[:ads]}} if split_path[:ads]
 
-              file_contents = on(host, powershell(command))&.stdout&.strip
+              file_contents = on(host, powershell(command, 'EncodedCommand' => true))&.stdout&.strip
```

Comment:
Would it be worth the added complexity of *only* encoding the command if we can find the magic cygwin tmp path?

Reply:
Well, another option is to just nail the temp directory to `C:\Windows\Temp`. I mean, it really doesn't matter where it is as long as it is consistent (I think).

Reply:
Though i dont know what we would search for.

Reply:
Ah, wait, that was the native cygwin magic that was getting in the way 🤦‍♂. Litmus seems to have gone the way of always Base64 encoding so that should work across the board.

Reply:
We could also just create our own directory at `C:\Windows\BeakerTemp`. Cygwin shouldn't mess with that.

Reply:
Just checked and the only call to `%SYSTEMROOT%` is the one in `beaker/host/windows/file.rb`. As long as things are using the `system_temp_path` function, we can set it to whatever we like.

Reply:
What if instead of relying on `sytem_temp_path` in https://github.com/puppetlabs/beaker-puppet/blob/50210ee3eeb769c4553d5e6a164bc7f093f81430/lib/beaker-puppet/install_utils/windows_utils.rb#L59-L61 we used https://github.com/puppetlabs/beaker/blob/74cd756b3673d9cf446529c1e1a61e45c64d7d2f/lib/beaker/host/windows/file.rb#L4-L10 . Can you run msi-exec from a cygwin path?

Reply:
I guess as long as you can be sure that `cygwin` is installed wherever you're going.

Most images do not have cygwin installed at this point.

Reply:
ahh right. So your `C:\Windows\BeakerTemp` idea, we would try to ensure that is created? then test authors would have to ensure that it is cleaned up? I'm not sure how cleanup goes in windows for stuff under temp, but creating a new directory under C:\\Windows seems like it could be problematic.

Reply:
Nah, directories are directories. It's cruft but `beaker` tears everything down at the end anyway.

Bolt would need to be careful, but `beaker` creates dynamically named tempfiles for everything that it runs in general anyway.

I figure that we would just see if the directory exists and create it if it doesn't. You can then bind the tempdir location into the SUT object itself so that it carries it around and just return it at every call to the function so that we don't hit the system every time.

(If anyone knows otherwise regarding the directories, please chime in!)